### PR TITLE
dts: Generate GPIO defines in non GPIO devices

### PR DESCRIPTION
**Pull request by erwango**
This pull request aim at generating #define for gpio properties defined in non gpio nodes.
For instance:
```
spbtle-rf@0 {
 	compatible = "st,spbtle-rf";
    	reset-gpios = <&gpioa 8 0>;
};
```
will trigger generation of:
```
#define ST_STM32_SPI_FIFO_40003C00_ST_SPBTLE_RF_0_RESET_GPIOS_CONTROLLER	"GPIOA"
#define ST_STM32_SPI_FIFO_40003C00_ST_SPBTLE_RF_0_RESET_GPIOS_FLAGS		0
#define ST_STM32_SPI_FIFO_40003C00_ST_SPBTLE_RF_0_RESET_GPIOS_PIN		8
```

This is done by introducing a new value in yaml dts bindings attribute "generation": 'define, use-prop-name'. If use-prop-name is set, #define will use property name as cell string instead of property generic name.
To achieve this, I've removed the 'cell-string' yaml attribute. It was used mostly to provide a generic value (IRQ, PINMUX, ..) and only once for a specific case. I enforced the generic value to be used directly in the script to simplify whole generation mechanism.

Last, I updated NXP dts files to use generated _CLOCK_CONTROLLER define.

I also took the opportunity to clean up from the script 'use-name-prefix' which was never used.

This PR is based on #7197

### CODE REVIEW

**Review by GAnthony — APPROVED** [review]
Since changes to TI dts.fixup files same as in #7197 - this looks good also.

File: dts/bindings/bluetooth/st,spbtle-rf.yaml [hunk 1]
```diff
@@ -18,4 +18,18 @@ properties:
     compatible:
       constraint: "st,spbtle-rf"
 
+    cs-gpios:
```
[the remark this answers is not in the document]
Comment (galak) [reply dts/bindings/bluetooth/st,spbtle-rf.yaml]: Hmm, from a device tree binding point of view, this seems to have been done as on the controller not the device side.

[the remark this answers is not in the document]
Comment (MaureenHelm) [reply dts/bindings/bluetooth/st,spbtle-rf.yaml]: I don't quite follow. Are you suggesting a particular change? Both st,spbtle-rf and nxp,mcr20a are users of the spi api, which allows specifying a gpio-controlled chip select.

[the remark this answers is not in the document]
Comment (galak) [reply dts/bindings/bluetooth/st,spbtle-rf.yaml]: If you look at https://git.kernel.org/pub/scm/linux/kernel/git/devicetree/devicetree-rebasing.git/tree/Bindings/spi/spi-bus.txt

The cs-gpios property is for the bus controller.  We use the reg property of the spi device as the CS number (ie index in array of cs-gpios - if gpios are used for CS on the controller.)

File: drivers/bluetooth/hci/Kconfig [hunk 2]
```diff
@@ -84,37 +84,41 @@ config BT_SPI_BLUENRG
 
 config BT_SPI_CHIP_SELECT_DEV_NAME
 	string "Chip Select (CS) line driver name"
-	depends on BT_SPI_BLUENRG
+	depends on BT_SPI_BLUENRG && !HAS_DTS_SPI_PINS
```
[the remark this answers is not in the document]
Comment (galak) [reply drivers/bluetooth/hci/Kconfig]: I fixed this up.

**Comment by galak** [comment]
recheck

**Pull request merged**